CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing In General
+So, you've come here hoping to make Concourse better?  Great!
+Concourse welcomes pull-request on any one of the many open-source
+pieces we currently have within our
+[organization](https://github.com/concourse).
+
+There are only a few ground rules that we like to see respected for
+your pull-request to be considered:
+
+- Make sure that you submit all pull-requests to the develop branch on
+Concourse (they won't build otherwise).
+- Each corresponding change should have an associated test to go along
+with it.  If you are having trouble testing, open an issue to discuss
+it.
+- Please don't forget to update the Concourse 
+[documentation](https://github.com/concourse/concourse/tree/develop/docs) 
+if you make a change to any the behavior (especially in fly).
+- Double check that all of the tests you have written pass on the
+pull-request
+
+With those ground rules out of the way, let's get you setup to work on
+the project!
+
+# Setup
+
+### Tools you will need
```

Comment:
Might be good to add a note on why each tool is needed, so they know why they're loading their system up with a bunch of tools. For tiny PRs they may only need a subset.